app/services/catalog/share_info.rb

```diff
@@ -15,26 +16,16 @@ def process
         ace.group_uuid
       end
 
-      group_names = fetch_group_names(uuids.uniq)
+      group_names = @user_context.group_names(:filter => uuids)
       @result = group_permissions.each_with_object([]) do |(uuid, permissions), memo|
         if group_names.key?(uuid)
           memo << { :group_name => group_names[uuid], :group_uuid => uuid, :permissions => permissions }
         else
           Rails.logger.warn("Skipping group UUID: #{uuid} since its missing from RBAC service")
         end
       end
-      self
-    end
-
-    private
 
-    def fetch_group_names(uuids)
-      opts = {:limit => MAX_GROUPS_LIMIT, :uuid => uuids}
-      Insights::API::Common::RBAC::Service.call(RBACApiClient::GroupApi) do |api|
-        Insights::API::Common::RBAC::Service.paginate(api, :list_groups, opts).each_with_object({}) do |group, memo|
```

Comment:
@eclarizio This is in account scope, not in principal scope. Its a different call from the one that we need when asking for your current group memberships.

When we make this call as a Catalog Administrator doing the Sharing we are trying to fetch group names for specific group_uuids. In our database we only store the group_uuid but not the name. This call has been optimized recently to pass in the list of uuids so we only get the group names for the ones that we care about.

```
{"@timestamp":"2020-04-16T14:00:35.169992 ","hostname":"catalog-api-179-vnx2h","pid":25,"tid":"2af21c66a1c8","level":"debug","message":"ET    HON: performed EASY effective_url=http://rbac.rbac-ci.svc:8080/api/rbac/v1/groups/?limit=500\u0026offset=0\u0026uuid=b4a8f1fa-1e87-48e0-b9    1e-ea384db846c7%2C296eee56-2adb-4947-8e2f-c96667eec60b response_code=200 return_code=ok total_time=0.029587","request_id":"80e9bc2cd6344ee78405e00ebdd305ff"}
```
In the above case the Portfolio is shared with 2 groups in the account or tenant.

Reply:
@mkanoor Good catch, I'll adjust it to pass in the options or something so that the two calls are distinct.